File: go_server/go_objects/teacher.go
package go_objects

type Teacher struct {
	TeacherID int `json:"teacherid"`
	Name string `json:"teachername"`
	Grade string `json:"grade"`
	Arrived bool `json:"arrived"`
}

func NewTeacher(teacherID int, name string, grade string) *Teacher {
	return &Teacher{TeacherID: teacherID, Name: name, Grade: grade}
}

func (t *Teacher) GetTeacherID() int {
	return t.TeacherID
}

func (t *Teacher) GetName() string {
	return t.Name
}

func (t *Teacher) GetGrade() string {
	return t.Grade
}

File: flutter_ui/lib/flutter_objects/teacher.dart
/// A placeholder class that represents an entity or model.
class Teacher {
  
  final int id;
  final String name;
  final String grade;
  bool arrived = false;

  Teacher(this.id, this.name, this.grade, this.arrived);

  /// The factory constructor is used to create a new
  ///  instance of the teacher class from a JSON (map) object.
  factory Teacher.fromJson(Map<String, dynamic> json) {
    try {
      // print(json);
      return Teacher(
        json['teacherid'] as int,
        json['teachername'] as String,
        json['grade'] as String,
        json['arrived'] as bool,
      );
    } catch (e) {
      throw const FormatException('Invalid or missing data in JSON');
    }
  }
  Map<String, dynamic> toJson() {
    return {
      'teacherid': id,
      'teachername': name,
      'grade': grade,
      'arrived': arrived,
    };
  }
}

class Teachers {
  final List<Teacher> teachers;

  Teachers({required this.teachers});

  factory Teachers.fromJson(List<dynamic> json) {
    List<Teacher> teachers = [];
    teachers = json.map((i) => Teacher.fromJson(i)).toList();
    return Teachers(teachers: teachers);
  }
}
